services/messaging/producer.js

```diff
@@ -0,0 +1,33 @@
+const { Kafka } = require("kafkajs");
+const logger = require("../../logs/logger");
+
+const topic = "email";
+
+const kafka = new Kafka({
+  clientId: "lms",
+  brokers: ["localhost:9092"],
+});
+
+const producer = kafka.producer();
+
+const connect = async () => {
+  try {
+    await producer.connect();
+  } catch (connectionError) {
+    logger.error("Error connecting to kafka :", connectionError);
+  }
+};
+
+const produce = async (message) => {
+  const serializedMessage = JSON.stringify(message);
+  try {
+    await producer.send({
+      topic,
+      messages: [{ value: serializedMessage }],
+    });
+  } catch (producerError) {
+    logger.error("Error producing message :", producerError);
+  }
+};
+
+module.exports = { produce, connect };
```

Comment:
Don't export connect but only produce. I don't think you need it for anything again. A connection is enough